src/data.ts

```diff
@@ -12,32 +21,61 @@ export default {
   ExternalERC721: {
     ledger: env.VITE_SC_EXTERNAL_ERC721_LEDGER_CONTRACT_ADDRESS,
     title: 'Mainnet NFT derivatives',
+    proofTitle: 'Mainnet NFTs',
     ownerTitle: 'This wallet owns a {derivative}',
     ownerContent:
       'This is a zkNFT derivative. It means this person has been verified to own at least one ‘{original}‘ Mainnet NFT.',
     proofIcon: Erc721Badge,
-
     badgeType: BadgeSourceType.ERC721,
     network: Network.Mainnet,
+    mint: async (provider: Web3Provider, proof: Proof) => {
```

Comment:
Maybe we need to extract this function into `helpers/contract`, like you did with the others?

Reply:
I'm not sure if this is needed